build-tools/packages/build-infrastructure/src/test/fluidRepo.test.ts

```diff
@@ -104,3 +108,24 @@ describe("loadFluidRepo", () => {
 		});
 	});
 });
+
+describe("setDependencyRange", () => {
+	const repo = loadFluidRepo(testRepoRoot);
+	const main = repo.releaseGroups.get("main" as ReleaseGroupName);
+	assert(main !== undefined);
+
+	const group2 = repo.releaseGroups.get("group2" as ReleaseGroupName);
+	assert(group2 !== undefined);
+
+	afterEach(async () => {
+		await git.checkout(["HEAD", "--", testRepoRoot]);
+		repo.reload();
+	});
+
+	it("updates the dependency range", async () => {
```

Comment:
Additional test cases to consider:

- Set the range to an explicit workspace interdependency range
- Set the range to a specific version

This is a fairly low-level command so I think the more interesting test cases will come later when we start building more stuff on top of this.